integreat_cms/core/utils/machine_translation_api_client.py

```diff
@@ -63,6 +76,40 @@ def translate_object(self, obj: Event | Page | POI, language_slug: str) -> None:
         """
         return self.translate_queryset([obj], language_slug)
 
+    def check_usage(
+        self,
+        region: Region,
+        source_translation: EventTranslation | (PageTranslation | POITranslation),
+    ) -> tuple[bool, int]:
+        """
+        This function checks if the attempted translation would exceed the region's word limit
+
+        :param region: region for which to check usage
+        :param source_translation: single content object
+        :return: translation would exceed limit, region budget, attempted translation word count
+        """
+        # Gather content to be translated and calculate total word count
+        attributes = [
+            getattr(source_translation, attr, None)
+            for attr in self.translatable_attributes
+        ]
+        content_to_translate = [
+            unescape(strip_tags(attr)) for attr in attributes if attr
+        ]
+
+        content_to_translate_str = " ".join(content_to_translate)
+        for char in "-;:,;!?\n":
+            content_to_translate_str = content_to_translate_str.replace(char, " ")
+        word_count = len(content_to_translate_str.split())
+
+        # Check if translation would exceed DeepL usage limit
+        region.refresh_from_db()
+        # Allow up to MT_SOFT_MARGIN more words than the actual limit
+        word_count_leeway = max(1, word_count - settings.MT_SOFT_MARGIN)
+        translation_exceeds_limit = region.deepl_budget_remaining < word_count_leeway
```

Comment:
```suggestion
        translation_exceeds_limit = region.mt_budget_remaining < word_count_leeway
```

(can't make a corresponding suggestion in the region model )